exec-service-wasmer/src/wasmer_metering.rs

```diff
@@ -76,48 +69,70 @@ impl ModuleMiddleware for Metering {
     fn transform_module_info(&self, module_info: &mut ModuleInfo) {
         let mut global_indexes = self.global_indexes.lock().unwrap();
 
-        let points_limit_global_index = module_info
-            .globals
-            .push(GlobalType::new(Type::I64, Mutability::Var));
-
-        module_info
-            .global_initializers
-            .push(GlobalInit::I64Const(self.points_limit as i64));
+        let mut create_global_index = |key: &str, init: i64| {
+            let global_index = module_info
+                .globals
+                .push(GlobalType::new(Type::I64, Mutability::Var));
 
-        module_info.exports.insert(
-            METERING_POINTS_LIMIT.to_string(),
-            ExportIndex::Global(points_limit_global_index),
-        );
+            module_info
+                .global_initializers
+                .push(GlobalInit::I64Const(init));
 
-        let points_used_global_index = module_info
-            .globals
-            .push(GlobalType::new(Type::I32, Mutability::Var));
+            module_info
+                .exports
+                .insert(key.to_string(), ExportIndex::Global(global_index));
 
-        module_info
-            .global_initializers
-            .push(GlobalInit::I32Const(0));
+            global_index
+        };
 
-        module_info.exports.insert(
-            METERING_POINTS_USED.to_string(),
-            ExportIndex::Global(points_used_global_index),
-        );
+        let points_limit = self.points_limit as i64;
```

Comment:
You probably meant to create a separate method which returns the `global_index`. This is too large to be an anonymous function.

Reply:
done